view/frontend/web/js/recobuy.js

```diff
@@ -35,56 +35,78 @@
 
 define([
     'nostojs',
-    'jquery',
-    'catalogAddToCart'
+    'jquery'
 ], function (nostojs, $) {
     'use strict';
 
-    //noinspection SpellCheckingInspection
-    var form = $('#nosto_addtocart_form').catalogAddToCart({});
     var Recobuy = {};
+
     Recobuy.addProductToCart = function (productId, element, quantity) {
         quantity = quantity || 1;
         var productData = {
-            "productId" : productId,
-            'skuId' : productId
+            productId : productId,
+            skuId : productId,
+            quantity: quantity
         };
-        Recobuy.addSkuToCart(productData, element, quantity);
+        return Recobuy.addSkuToCart(productData, element);
     };
 
     // Products must be and array of objects [{'productId': '123', 'skuId': '321'}, {...}]
     // skuId is optional for simple products.
     Recobuy.addMultipleProductsToCart = function (products, element) {
         if (Array.isArray(products)) {
-            products.forEach(function (productObj) {
-                Recobuy.addSkuToCart(productObj, element, 1);
-            });
+            return Recobuy.recursiveAddProducts(products, element)
+        } else {
+            Promise.reject(new Error("Products is not type array"))
         }
     };
 
+    Recobuy.recursiveAddProducts = function (products, element) {
+        var length = products.length;
+        if (length === 0) {
+            return Promise.resolve();
+        }
+        return Recobuy.addSkuToCart(products[0], element, 1)
+            .then(function (res) {
+                return Recobuy.recursiveAddProducts(products.slice(1), element)
+            })
+    }
+
     // Product object must have fields productId and skuId {'productId': '123', 'skuId': '321'}
-    Recobuy.addSkuToCart = function (product, element, quantity) {
-        quantity = quantity || 1;
+    Recobuy.addSkuToCart = function (product, element) {
+
+        var quantity = product.quantity || 1;
+        var url = document.querySelector("#nosto_addtocart_form").getAttribute("action")
+        var formKey = document.querySelector("#nosto_addtocart_form > input[name='form_key']").getAttribute("value")
+
+
+        return new Promise(function (resolve, reject) {
+            $.post(url, {
+                form_key: formKey,
+                qty: quantity,
+                product: product.productId,
+                sku: product.skuId
+            }).done(function() {
+                Recobuy.sendCartEvent(element, product.productId)
+                return resolve()
```

Comment:
Do you need to return or simply resolve()?

Reply:
Return a promise